src/test/java/org/broadinstitute/hellbender/utils/fragments/FragmentUtilsUnitTest.java

```diff
@@ -76,4 +76,110 @@ public void testAdjustingTwoReads(final GATKRead read1, final GATKRead read2, fi
             Assert.assertEquals(read2.getBaseQualities()[i], HIGH_QUALITY);
         }
     }
+
+
+    // Generate a bunch of reads with softclips that do not overlap with the other read.
+    @DataProvider(name = "AdjustFragmentsTestSoftClipsNotOverlapping")
+    public Object[][] createAdjustFragmentsTestSoftClips() throws Exception {
+        List<Object[]> tests = new ArrayList<>();
+
+        final String leftFlank = "CCC";
+        final String rightFlank = "AAA";
+        final String allOverlappingBases = "ACGTACGTGGAACCTTAG";
+        for ( int overlapSize = 1; overlapSize < allOverlappingBases.length(); overlapSize++ ) {
+            for (int leftSoftclip = 0; leftSoftclip <= leftFlank.length(); leftSoftclip++) {
+                for (int rightSoftclip = 0; rightSoftclip <= rightFlank.length(); rightSoftclip++) {
+                    final String overlappingBases = allOverlappingBases.substring(0, overlapSize);
+                    final byte[] overlappingBaseQuals = new byte[overlapSize];
+                    for ( int i = 0; i < overlapSize; i++ ) {
+                        overlappingBaseQuals[i] = HIGH_QUALITY;
+                    }
+                    final GATKRead read1  = makeOverlappingRead(leftFlank, HIGH_QUALITY, overlappingBases, overlappingBaseQuals, "", HIGH_QUALITY, 1, leftSoftclip, 0);
+                    final GATKRead read2  = makeOverlappingRead("", HIGH_QUALITY, overlappingBases, overlappingBaseQuals, rightFlank, HIGH_QUALITY, leftFlank.length() + 1, 0, rightSoftclip);
+                    tests.add(new Object[]{read1, read2, overlapSize});
+                }
+            }
+        }
+        return tests.toArray(new Object[][]{});
+    }
+
+    // Assert that despite the softclips that the reads are being properly
+    @Test(dataProvider = "AdjustFragmentsTestSoftClipsNotOverlapping")
+    public void testAdjustingTwoReadsWithSoftClipping(final GATKRead read1, final GATKRead read2, final int overlapSize) {
```

Comment:
Could you put in a test with indels with a comment that the behavior is not yet ideal?  It would help future efforts by clarifying exactly which cases currently don't work right.